sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs

```diff
@@ -110,6 +110,17 @@ private static void CheckTypeName(SyntaxNodeAnalysisContext context)
                 return;
             }
 
+            if (typeDeclaration is ClassDeclarationSyntax)
+            {
+                var className = typeDeclaration.Identifier.ValueText;
+                if (className != "ITest"
```

Comment:
nitpick - I think the checks can be in a private  method / local function to make this `if` easier to read